### ISSUE TITLE
[Feature Request] Filter Models concepts tree based on a 'viewpoint' definition

### ISSUE THREAD

**Issue by AlexisHFr** [archimatetool/archi#1078]
### Description

Working on a large Models with thousands of concepts requires to spend quite a lot of time searching and filtering these concepts.

One topic where Archi could help a bit further : enforce filter based on a  viewpoint definition 

eg : 
  - if I create a new view with viewpoint 'Application Cooperation', the Models tree view could automaticaly apply a filter on the same concept types as the Palette view : here only Application concepts and Location/Group.
  - such a filter could be selected by the user as an option labeled 'dynamic viewpoint filtering', 
  - drawback will be a performance issue if the user repetively change the active view.. so maybe a static filter definition based on a viewpoint selection would be better / it won't need to track current active view and could be a good tradeoff.

**Comment by Phillipus** [comment]
> drawback will be a performance issue

Indeed. The Models Tree is resource heavy and updating and refreshing all of the tree nodes is an expensive operation. That's why the current implementation greys out the concepts rather than hiding them.

**Comment by AlexisHFr** [comment]
From my experience the Models tree is hiding elements based on selected filters , and there is no perf issue even with a 6K concepts tree, but I understand it would be wise to trigger such a tree refresh automaticaly / when the $current-view.viewpoint state changed.

I think to overcome performance issue the action should be triggered by the user each time he wants a re-filtering

So my proposal is to add a menu item in the Filters drop-down menu as a shortcut to apply viewpoints-based filter rather than manualy selecting 4-10 concept types in the filter tree.
